scripts/specs/test/CTestPuppet.lua

```diff
@@ -0,0 +1,59 @@
+---@meta
+
+-- Test puppet that combines client and entity functionality
```

Comment:
I feel like `puppet` is an overloaded term in the context of this codebase. My first thought was "why do automatons need their own thing". Can we come up with something better? Even if it's verbose like `ClientEntityPair` (just an example)